File: 242-valid-anagram/valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
		return false
	}
    
	myMap_s := make(map[string]int)
	myMap_t := make(map[string]int)

	for index := range s {
		//s
		if _, exists := myMap_s[string(s[index])]; exists {
			myMap_s[string(s[index])] += 1
		} else {
			myMap_s[string(s[index])] = 1
		}

		//t
		if _, exists := myMap_t[string(t[index])]; exists {
			myMap_t[string(t[index])] += 1
		} else {
			myMap_t[string(t[index])] = 1
		}
	}

	for key, value1 := range myMap_s {
		value2, exists := myMap_t[key]
		if !exists || value1 != value2 {
			return false
		}
	}
	return true
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        else:
            first = {}
            second = {}
            for i in range(len(s)):
                if s[i] not in first:
                    first[s[i]] = 1
                elif s[i] in first:
                    first[s[i]] += 1
                if t[i] not in second:
                    second[t[i]] = 1
                elif t[i] in second:
                    second[t[i]] += 1

            if first == second:
                return True
            else:
                return False